front/app/components/FormBuilder/components/FormFields/FormField/Logic/index.tsx

```diff
@@ -43,6 +43,18 @@ const Logic = ({
 
   const pageMessage = formatMessage(messages.page);
 
+  // Add the additional catch all logic options
+  const catchAllTypes = field.required
+    ? ['any_other_answer']
+    : ['any_other_answer', 'no_answer'];
+  const catchAllLogicRules = field.logic.rules
+    ?.filter((rule) => catchAllTypes.includes(rule.if.toString()))
```

Comment:
Minor and optional
```suggestion
  const catchAllLogicRules = (field.logic.rules || [])
    .filter((rule) => catchAllTypes.includes(rule.if.toString()))
```

If we default to an empty array here, we can remove this check `catchAllLogicRules &&` on line 136.